### PR TITLE
add the build parameter to support submitting the private registry ac…

### PR DESCRIPTION
**Pull request by lllwan**
…count password

### CODE REVIEW

**Comment by justadogistaken** [comment]
@lllwan And we can read from ~/.docker/config.json to see if the auth info already exists.

File: build/lite/docker/pull.go [hunk 1]
```diff
@@ -60,27 +63,38 @@ func (d Docker) ImagesPullByList(images []string) {
...
+		logger.Warn("docker client creation failed: %v", err)
 		return err
 	}
-	if d.Username != "" && d.Password != "" {
-		authConfig := types.AuthConfig{
-			Username: d.Username,
-			Password: d.Password,
-		}
+	authConfig, err = utils.GetDockerAuthInfoFromDocker(named.Domain())
+	if err != nil {
```
Comment (justadogistaken) [inline build/lite/docker/pull.go:86]: err != nil ?

Comment (justadogistaken) [reply build/lite/docker/pull.go:86]: I think we can tolerate the fault here, because we could mount config into registry.

Comment (lllwan) [reply build/lite/docker/pull.go:88]: OK

File: build/lite/docker/pull.go [hunk 2]
```diff
@@ -60,27 +63,37 @@ func (d Docker) ImagesPullByList(images []string) {
...
-			Username: d.Username,
-			Password: d.Password,
-		}
-		encodedJSON, err := json.Marshal(authConfig)
-		if err != nil {
-			return err
-		}
-		authStr := base64.URLEncoding.EncodeToString(encodedJSON)
-		ImagePullOptions = types.ImagePullOptions{RegistryAuth: authStr}
+	authConfig, _ = utils.GetDockerAuthInfoFromDocker(named.Domain())
```
Comment (justadogistaken) [inline build/lite/docker/pull.go:86]: we still need the error, if error happens, just pass a empty RegistryAuth.

Comment (lllwan) [reply build/lite/docker/pull.go:86]: The GetDockerAuthInfoFromDocker function handles the error and must return an auth object

**Review by justadogistaken — APPROVED** [review]

**Pull request merged**